src/dotnet/SlnFileExtensions.cs

```diff
@@ -0,0 +1,310 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Construction;
+using Microsoft.Build.Evaluation;
+using Microsoft.Build.Execution;
+using Microsoft.DotNet.Cli.Sln.Internal;
+using Microsoft.DotNet.Cli.Utils;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.DotNet.Tools.Common
+{
+    public static class SlnFileExtensions
+    {
+        public static void AddProject(this SlnFile slnFile, string fullProjectPath)
```

Comment:
Arguments should be validated, e.g. it's preferable to throw `ArgumentNullException` rather than `NullReferenceException`.